compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -710,6 +742,26 @@ private BIRVariableDcl getSelf(BSymbol receiver) {
         return self;
     }
 
+    private void analyzeParametersAndReturnType(List<BLangSimpleVariable> requiredParams, BLangVariable restParam,
+                                                BLangType returnTypeNode) {
+        BLangType typeNode;
+        for (BLangSimpleVariable parameter : requiredParams) {
+            typeNode = parameter.typeNode;
+            if (typeNode != null) {
+                typeNode.accept(this);
+            }
```

Comment:
It seems like this `null` check and visiting happens frequently. How about extracting this logic to a new function and call it?
```java
analyze(paramter.typenode)

private void analyze(BLangType node) {
    if (node != null) {
        node.accept(this);
    }
}
```